BAMF FDG-Avid Breast Tumor

jithenece opened the pull request:
AIMI1 - [Pretrained model](https://zenodo.org/records/8290055) for 3D semantic image segmentation of the FDG-avid lesions from PT/CT scans

jithenece committed:
create multi modality PET CT nnUnet Runner

jithenece committed:
add Resampler module

jithenece committed:
define workflow

jithenece committed:
generate Total Segmentator organs

jithenece committed:
limiting tumor to breast

jithenece committed:
add custom registration module

jithenece committed:
update metadata

jithenece committed:
use FDG_AVID_TUMOR entry

jithenece committed:
roi segments updated

jithenece committed:
use FDG_AVID_TUMOR label

jithenece committed:
more results stats

jithenece committed:
Merge branch 'MHubAI:main' into bamf_nnunet_pet_ct_breast

jithenece committed:
updated label and comments

jithenece committed:
reformat code

jithenece commented:
```yaml
sample:
  idc_version: Version 2: Updated 2020/01/10
  data:
  - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.8162.7003.200887946066796652452097013479
    aws_url: s3://idc-open-data/7a375cfa-8708-46f0-b056-248dbaca851e/*
    path: case1/ct
  - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.8162.7003.182230913012492714502249429083
    aws_url: s3://idc-open-data/739449f3-286c-4511-86a9-39172a50dc95/*
    path: case1/pt

reference:
  url: https://drive.google.com/file/d/1afkRFJqwgii1tUSMiQV_dPlDsUBl6_0g/view?usp=sharing
```

jithenece commented:
[Archive.zip](https://github.com/user-attachments/files/16069186/Archive.zip)
Slicer screenshots added

LennyN95 commented on models/bamf_pet_ct_breast_tumor/utils/Registration.py:
```diff
@@ -0,0 +1,58 @@
+        resampler.SetDefaultPixelValue(int(np.min(sitk.GetArrayFromImage(moving))))
+        resampler.SetTransform(outTx)
+        out = resampler.Execute(moving)
+        tmp_dir = self.config.data.requestTempDir(label="registration-processor")
+        output_path = os.path.join(tmp_dir, f'registered.nii.gz')
+        out.CopyInformation(fixed)
+        sitk.WriteImage(out, output_path)
+        shutil.copyfile(output_path, out_data.abspath)
```
Instead of copying the file, it `out_data.abspath` could be used directly and replace `output_path`.

jithenece replied:
I have updated the logic

LennyN95 commented on models/bamf_pet_ct_breast_tumor/config/default.yml:
```diff
@@ -0,0 +1,49 @@
+
+execute:
+- FileStructureImporter
+- SitkNiftiConverter
```
Why do we need this converter? Cannot we use the [NiftiConverter](https://github.com/MHubAI/documentation/blob/main/documentation/mhubio/mhubio_modules.md#nifticonverter) module? There you can configure the backend to use dcm2niix which offers high performance.

jithenece replied:
i have changed engine to dcm2niix. it required upgrading `dcm2niix==1.0.20220715` to fix the conversion issue on few ct scans.

LennyN95 commented on models/bamf_pet_ct_breast_tumor/utils/BreastPostProcessor.py:
```diff
@@ -0,0 +1,114 @@
+        op_data = self.n_connected(op_data)
+        op_img = sitk.GetImageFromArray(op_data)
+        op_img.CopyInformation(ref)
+        tmp_dir = self.config.data.requestTempDir(label="breast-post-processor")
+        tmp_file = os.path.join(tmp_dir, f'final.nii.gz')
+        sitk.WriteImage(op_img, tmp_file)
+
+        shutil.copyfile(tmp_file, out_data.abspath)
```
See comments above about copying files.

jithenece replied:
i have updated the logic

LennyN95 commented on models/bamf_pet_ct_breast_tumor/utils/NNUnetPETCTRunner.py:
```diff
@@ -0,0 +1,215 @@
+
+            structure = structure_list[channel] if channel < len(structure_list) else f"structure_{channel}"
+            pred_softmax_segmask = pred_softmax_all[channel].astype(dtype = np.float32)
+
+            if output_dtype == "float32":
+                # no rescale needed - the values will be between 0 and 1
+                # set SITK image dtype to Float32
+                sitk_dtype = sitk.sitkFloat32
+
+            elif output_dtype == "uint8":
+                # rescale between 0 and 255, quantize
+                pred_softmax_segmask = (255*pred_softmax_segmask).astype(np.int32)
+                # set SITK image dtype to UInt8
+                sitk_dtype = sitk.sitkUInt8
+
+            elif output_dtype == "uint16":
+                # rescale between 0 and 65536
+                pred_softmax_segmask = (65536*pred_softmax_segmask).astype(np.int32)
+                # set SITK image dtype to UInt16
+                sitk_dtype = sitk.sitkUInt16
+            else:
+                raise ValueError("Invalid output data type. Please choose between uint8, uint16 or float32.")
+                
+            pred_softmax_segmask_sitk = sitk.GetImageFromArray(pred_softmax_segmask)
+            pred_softmax_segmask_sitk.CopyInformation(sitk_ct)
+            pred_softmax_segmask_sitk = sitk.Cast(pred_softmax_segmask_sitk, sitk_dtype)
+
+            # generate data
+            prob_mask = InstanceData(f'{structure}.nrrd', DataType(FileType.NRRD, {'mod': 'prob_mask', 'structure': structure}), bundle=prob_masks_bundle)
+
+            # export file
+            writer = sitk.ImageFileWriter()
+            writer.UseCompressionOn()
+            writer.SetFileName(prob_mask.abspath)
+            writer.Execute(pred_softmax_segmask_sitk)
+
+            # check if the file was written
+            if os.path.isfile(prob_mask.abspath):
+                self.v(f" > prob mask for {structure} saved to {prob_mask.abspath}")
+                prob_mask.confirm()
```
Did you test the probability map export under the modified circumstances?

LennyN95 commented on models/bamf_pet_ct_breast_tumor/config/default.yml:
```diff
@@ -0,0 +1,49 @@
+- SitkNiftiConverter
+- Registration
+- NNUnetPETCTRunner
+- TotalSegmentatorMLRunner
```
Why is TotalSegmentator run as part of this model?

LennyN95 commented on models/bamf_pet_ct_breast_tumor/config/default.yml:
```diff
@@ -0,0 +1,49 @@
+    nnunet_model: 3d_fullres
+    roi: LIVER,KIDNEY,URINARY_BLADDER,SPLEEN,LUNG,BRAIN,HEART,STOMACH,BREAST+FDG_AVID_TUMOR
+
+  TotalSegmentatorMLRunner:
+    in_data: nifti:mod=ct:registered=true
+    use_fast_mode: false
```
See above.

LennyN95 replied:
I'd assume that fast mode would work just as well but should significantly speed up the process.

LennyN95 commented on models/bamf_pet_ct_breast_tumor/config/default.yml:
```diff
@@ -0,0 +1,49 @@
+      - $patientID/pt@dicom:mod=pt
+    import_id: patientID
+
+  SitkNiftiConverter:    
+    in_datas: dicom:mod=pt|ct
+    allow_multi_input: true
```
See above.

LennyN95 commented on models/bamf_pet_ct_breast_tumor/dockerfiles/Dockerfile:
```diff
@@ -0,0 +1,33 @@
+# https://github.com/MIC-DKFZ/nnUNet/pull/1209
+ENV SKLEARN_ALLOW_DEPRECATED_SKLEARN_PACKAGE_INSTALL=True
+
+# Install nnunet
+# Install TotalSegmentator
+RUN pip3 install TotalSegmentator==1.5.7 nnunet==1.6.6
```
See above (why do we need TS for this model).

LennyN95 commented on models/bamf_pet_ct_breast_tumor/utils/TotalSegmentatorMLRunner.py:
```diff
@@ -0,0 +1,157 @@
+    'rib_right_1': 'RIGHT_RIB_1',
+    'rib_right_2': 'RIGHT_RIB_2',
+    'rib_right_3': 'RIGHT_RIB_3',
+    'rib_right_4': 'RIGHT_RIB_4',
+    'rib_right_5': 'RIGHT_RIB_5',
+    'rib_right_6': 'RIGHT_RIB_6',
+    'rib_right_7': 'RIGHT_RIB_7',
+    'rib_right_8': 'RIGHT_RIB_8',
+    'rib_right_9': 'RIGHT_RIB_9',
+    'rib_right_10': 'RIGHT_RIB_10',
+    'rib_right_11': 'RIGHT_RIB_11',
+    'rib_right_12': 'RIGHT_RIB_12',
+    'humerus_left': 'LEFT_HUMERUS',
+    'humerus_right': 'RIGHT_HUMERUS',
+    'scapula_left': 'LEFT_SCAPULA',
+    'scapula_right': 'RIGHT_SCAPULA',
+    'clavicula_left': 'LEFT_CLAVICLE',
+    'clavicula_right': 'RIGHT_CLAVICLE',
+    'femur_left': 'LEFT_FEMUR',
+    'femur_right': 'RIGHT_FEMUR',
+    'hip_left': 'LEFT_HIP',
+    'hip_right': 'RIGHT_HIP',
+    'sacrum': 'SACRUM',
+    'face': 'FACE',
+    'gluteus_maximus_left': 'LEFT_GLUTEUS_MAXIMUS',
+    'gluteus_maximus_right': 'RIGHT_GLUTEUS_MAXIMUS',
+    'gluteus_medius_left': 'LEFT_GLUTEUS_MEDIUS',
+    'gluteus_medius_right': 'RIGHT_GLUTEUS_MEDIUS',
+    'gluteus_minimus_left': 'LEFT_GLUTEUS_MINIMUS',
+    'gluteus_minimus_right': 'RIGHT_GLUTEUS_MINIMUS',
+    'autochthon_left': 'LEFT_AUTOCHTHONOUS_BACK_MUSCLE',
+    'autochthon_right': 'RIGHT_AUTOCHTHONOUS_BACK_MUSCLE',
+    'iliopsoas_left': 'LEFT_ILIOPSOAS',
+    'iliopsoas_right': 'RIGHT_ILIOPSOAS',
+    'urinary_bladder': 'URINARY_BLADDER'
+}
+
+# from totalsegmentator.map_to_binary import class_map
+# ROI = ','.join(mapping[class_map['total'][ci]] for ci in range(1, 105))
+ROI = 'SPLEEN,RIGHT_KIDNEY,LEFT_KIDNEY,GALLBLADDER,LIVER,STOMACH,AORTA,INFERIOR_VENA_CAVA,PORTAL_AND_SPLENIC_VEIN,PANCREAS,RIGHT_ADRENAL_GLAND,LEFT_ADRENAL_GLAND,LEFT_UPPER_LUNG_LOBE,LEFT_LOWER_LUNG_LOBE,RIGHT_UPPER_LUNG_LOBE,RIGHT_MIDDLE_LUNG_LOBE,RIGHT_LOWER_LUNG_LOBE,VERTEBRAE_L5,VERTEBRAE_L4,VERTEBRAE_L3,VERTEBRAE_L2,VERTEBRAE_L1,VERTEBRAE_T12,VERTEBRAE_T11,VERTEBRAE_T10,VERTEBRAE_T9,VERTEBRA ...
```
can be removed, right?

LennyN95 commented on models/bamf_pet_ct_breast_tumor/utils/BreastPostProcessor.py:
```diff
@@ -0,0 +1,114 @@
+        """
+        Perform postprocessing and writes simpleITK Image
+        """
+        tumor_seg_path = in_tumor_data.abspath
+        total_seg_path = in_total_seg_data.abspath
+
+        ts_data = sitk.GetArrayFromImage(sitk.ReadImage(total_seg_path))
+        ts_abdominal = sitk.GetArrayFromImage(sitk.ReadImage(total_seg_path))
+        ts_data[ts_data > 1] = 1
+        lesions = sitk.GetArrayFromImage(sitk.ReadImage(tumor_seg_path))
+        tumor_label = 9
+        lesions[lesions != tumor_label] = 0
+        lesions[lesions == tumor_label] = 1
+
+        op_data = np.zeros(ts_data.shape)
+        ref = sitk.ReadImage(in_ct_data.abspath)
+        ct_data = sitk.GetArrayFromImage(ref)
+
+        op_data[lesions == 1] = 1
+        th = np.min(ct_data)
+        op_data[ct_data == th] = 0  # removing predicitons where CT not available
+        # Use the coordinates of the bounding box to crop the 3D numpy array.
+        ts_abdominal[ts_abdominal > 4] = 0
+        ts_abdominal[ts_abdominal > 1] = 1
+        if ts_abdominal.max() > 0:
+            x1, x2, y1, y2, z1, z2 = self.bbox2_3D(ts_abdominal)
+        # Create a structuring element with ones in the middle and zeros around it
+        structuring_element = np.ones((3, 3))
```
Isn't running TS a bit overkill for the task? What about taking abdominal segmentation masks as an input to the pipeline? Then a user could run any model (e.g. TotalSegmentator from MHub) and provide the dicomseg file alongside the input dicom files. 

I#m asking this to evaluate the situation and to get the best possible user experience.

vanossj replied:
The simplest user experience would be to have dicom in -> dicom out. and not have a segmentation also be an input.

jithenece replied:
@LennyN95 please let us know if you are fine with this.

LennyN95 replied:
It'd be nice to provide an alternative workflow (config file) then, where you start with a DicomImporter expecting TotalSegmentator DICOMSEG files (e.g. as generated when running the MHub TotalSegmentator model) and use a DsegExtractor in the execute chain to extract the segmentations.

vanossj replied:
@jithenece an alternative workflow sounds great, then we have both options

jithenece replied:
@LennyN95 I have added `multi` flow which takes `total segmentator` files as input. please check and let me know for any changes.

jithenece replied:
I have changed the alternative workflow (config file) to `composite`. Please check

LennyN95 replied:
Looks good! Thank you @jithenece!

LennyN95 reviewed (changes requested):
Thank you for this implementation, this appears to be a more elaborate pipeline. Well done! I added some questions and discussion points in my review.

jithenece committed:
review comment fixes

jithenece committed:
update comments

jithenece commented:
/test

```yaml
sample:
  idc_version: "Data Release 2.0 January 10, 2020"
  data:
  - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.8162.7003.200887946066796652452097013479
    aws_url: s3://idc-open-data/7a375cfa-8708-46f0-b056-248dbaca851e/*
    path: 'case_study1/ct'
  - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.8162.7003.182230913012492714502249429083
    aws_url: s3://idc-open-data/739449f3-286c-4511-86a9-39172a50dc95/*
    path: 'case_study1/pt'

reference:
  url: https://drive.google.com/file/d/1yivXqTBMXslsmj3uD8gE9LojLCKh-BqV/view?usp=sharing
```

LennyN95 commented:
@jithenece I wanted to run our test routine on all read for testing PR's, however, I noticed, the sample reference url you provided won't work with `wget -O /tmp/ref.zip $RESSOURCE_URL`. This might be specific to using a google drive link, however, I generally advice you to instead to attach your data to the PR directly (e.g., by uploading them to the comment) and to use the permanent link as a reference. 

Please note, this needs to be updated on all effected PRs until testing can proceed.

jithenece committed:
update model name

jithenece committed:
Merge branch 'MHubAI:main' into bamf_pet_ct_breast_tumor

jithenece committed:
multi workflow for tseg file as input

jithenece committed:
format code

jithenece committed:
update body part

jithenece commented:
/test 

Not many collection available for breast-pet-ct.

attaching segmentation
[output.zip](https://github.com/user-attachments/files/16927110/output.zip)

```yaml
sample: 
  idc_version: 17.0
  data:
  - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.8162.7003.539267076861125410814830191835
    aws_url: s3://idc-open-data/84c9b972-76e1-4fa3-a7ea-19b6500e497a/*
    path: case_study1/ct
  - SeriesInstanceUID: 1.3.6.1.4.1.14519.5.2.1.8162.7003.196821690630879561473146713439
    aws_url: s3://idc-open-data/33f0bd14-4bf5-469c-83db-183e1ab96f02/*
    path: case_study1/pt

reference:
  url: https://github.com/user-attachments/files/16927110/output.zip
```

LennyN95 commented:
Please note, we **updated our base image**. All mhub dependencies are now installed in a virtual environment under `/app/.venv` running `Python 3.11`. Python, virtual environment and dependencies are now managed with [uv](https://github.com/astral-sh/uv). If required, you can create custom virtual environments, e.g., `uv venv -p 3.8 .venv38` and use `uv pip install -p .venv38 packge-name` to install dependencies and `uv run -p .venv3.8 python script.py` to run a python script. 

We also **simplified our test routine**. Sample and reference data now have to be uploaded to Zenodo and provided in a mhub.tom file at the project root. The process how to create and provide these sample data is explained in the updated [testing phase article](https://github.com/MHubAI/documentation/blob/main/documentation/mhub_contribution/testing_phase.md) of our documentation. Under [doi.org/10.5281/zenodo.13785615](https://doi.org/10.5281/zenodo.13785615) we provide sample data as a reference.

jithenece committed:
Merge branch 'MHubAI:main' into bamf_pet_ct_breast_tumor

jithenece committed:
update publication link

jithenece committed:
/test validation

LennyN95 committed:
Update Dockerfile

- pin torch to v2.2.2

LennyN95 commented:
![Static Badge](https://img.shields.io/badge/Passed-x?style=flat&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CCjxzdmcgd2lkdGg9IjQxcHgiIGhlaWdodD0iNDJweCIgdmlld0JveD0iMCAwIDQxIDQyIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPgogICAgPGcgaWQ9IndlYnNpdGUiIHN0cm9rZT0ibm9uZSIgc3Ryb2tlLXdpZHRoPSIxIiBmaWxsPSJub25lIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiPgogICAgICAgIDxwYXRoIGQ9Ik01LjAzMiwwLjgyODU0MTc0MiBDNi4wNTg0MzQ2MSwwLjgyODU0MTc0MiA3LjA1NDUyMzM4LDEuMTc2NjcwNzggNy44NTc1MDAwNywxLjgxNjA0MjcxIEw3Ljg1NzUwMDA3LDEuODE2MDQyNzEgTDIwLjk5NiwxMi4yNzc1NDE3IEwzMy42NTQxOTc2LDIuMTk5MDI0MTUgQzM0LjMwMjAwNzIsMS42ODMyMDQzOCAzNS4wOTI2MjYyLDEuMzgwODM4MTEgMzUuOTE2NjQxOSwxLjMzMDEyNTQgTDM2LjE2NDczOTIsMS4zMjI0OTkzOSBDMzguMzE3MDE3NSwxLjMyMjQ5OTM5IDQwLjA3NDEzMjcsMy4wMDY0Njc4NiA0MC4xNzg2NDgyLDUuMTIyODU5MDUgTDQwLjE4MzU2NjUsNS4zMjI0OTkzOSBMNDAuMTgzNTY2NSwzNi42Nzc1MDA2IEM0MC4xODM1NjY1LDM3LjU4NTc4NjYgMzkuODcyOTg4NSwzOC40NjcwMjg2IDM5LjMwMjkxNjEsMzkuMTc2MjgwOCBDMzcuOTYyNjAwMSw0MC44NDM4MjY5IDM1LjU1NDY5NjYsNDEuMTYwNzYyNCAzMy44MzAwOTU0LDM5LjkzMzQ1NTIgTDMzLjY1NDE5NzYsMzkuODAwOTc1OCBMMjAuOTk3LDI5LjcyMDU0MTcgTDcuODU3NTAwMDcsNDAuMTgzOTU3MyBDNS45NTcxMDU1LDQxLjY5NzE1MDYgMy4yMTYxMTQ0LDQxLjQzODYyODEgMS42Mjg4NjQ2NywzOS42MzQ2MDYzIEwxLjQ4MzUwMDk3LDM5LjQ2MDk1ODMgQzAuODQ0MTI5MDQsMzguNjU3OTgxNiAwLjQ5NiwzNy42NjE4OTI5IDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDUuMzY0NTQxNzQgQzAuNDk2LDIuODU5Mzc4MTIgMi41MjY4MzYzNywwLjgyODU0MTc0MiA1LjAzMiwwLjgyODU0MTc0MiBaIE01LjAzMiwxLjgzNjU0MTc0IEMzLjA4MzUzOTQsMS44MzY1NDE3NCAxLjUwNCwzLjQxNjA4MTE0IDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDM2LjYzNTQ1ODMgQzEuNTA0LDM3LjQzMzc5NjMgMS43NzQ3NjcwMywzOC4yMDg1MzIgMi4yNzIwNTYzMSwzOC44MzMwNjk0IEMzLjQ4NTc2MzQ0LDQwLjM1NzM0NDIgNS43MDUzMzYzNSw0MC42MDkxMDkxIDcuMjI5NjExMTYsMzkuMzk1NDAxOSBMNy4yMjk2MTExNiwzOS4zOTU0MDE5IEwyNi44NjU5MjU5LDIzLjc1OTk0MzcgQzI3LjA3Mzk2NDcsMjMuNTk0MjkyMyAyNy4yNjI2MDcsMjMuNDA1NjUgMjcuNDI4MjU4NCwyMy4xOTc2MTEyIEMyOC42NDE5NjU1LDIxLjY3MzMzNjQgMjguMzkwMjAwNywxOS40NTM3NjM0IDI2Ljg2NTkyNTksMTguMjQwMDU2MyBMMjYuODY1OTI1OSwxOC4yNDAwNTYzIEw3LjIyOTYxMTE2LDIuNjA0NTk4MDUgQzYuNjA1MDczNzQsMi4xMDczMDg3NyA1LjgzMDMzODAzLDEuODM2NTQxNzQgNS4wMzIsMS44MzY1NDE3NCBaIE01LjU0OTY0NDc2LDUuMTQyNjMxMDMgTDUuNjQxNjIxMyw1LjIwNzQxNDU2IEwyNC43MzEwNTExLDIwLjIwNzQxNDYgQzI1LjE2ODc4NCwyMC41NTEzNzQyIDI1LjI0NDgwMjIsMjEuMTg1MDYxMSAyNC45MDA4NDI2LDIxLjYyMjc5NCBDMjQuNTgzMzQxNCwyMi4wMjY4NTUxIDI0LjAxODk3MiwyMi4xMjI3MDk2IDIzLjU4OTczNTksMjEuODY0NTMyIEwyMy40ODU0NjMyLDIxLjc5MjU4NTQgTDYuMDI3LDguMDc0IEw2LjAyNjgyNzMzLDM1IEM2LjAyNjgyNzMzLDM1LjUxOTU4OTUgNS42MzM2OTc1MywzNS45NDczNTg2IDUuMTI4NjYwMDQsMzYuMDAyMDg1MiBMNS4wMTg4MjczMywzNi4wMDggQzQuNDk5MjM3ODQsMzYuMDA4IDQuMDcxNDY4NzQsMzUuNjE0ODcwMiA0LjAxNjc0MjE0LDM1LjEwOTgzMjcgTDQuMDEwODI3MzMsMzUgTDQuMDEwODI3MzMsNiBDNC4wMTA4MjczMyw1LjE5NzEwODMzIDQuODk0MTIyODQsNC43MzEzMjEwNCA1LjU0OTY0NDc2LDUuMTQyNjMxMDMgWiIgaWQ9IkNvbWJpbmVkLVNoYXBlIiBmaWxsPSIjRkZGRkZGIj48L3BhdGg%2BCiAgICA8L2c%2BCjwvc3ZnPg%3D%3D&logoColor=blue&label=Model%20Test&labelColor=%23313F49)

<details><summary>Test Results</summary>

```yaml
id: 9e3f0bef-d079-4904-a60f-20ac5a18e2f2
name: MHub Test Report (default)
date: '2025-02-25 16:39:48'
checked_files:
- file: bamf_pet_ct_breast_tumor.seg.dcm
  path: /app/data/output_data/case1/bamf_pet_ct_breast_tumor.seg.dcm
  checks:
  - checker: DicomsegContentCheck
    notes:
    - label: Segment Count
      description: The number of segments identified in the inspected dicomseg file.
      info: 1
summary:
  files_missing: 0
  files_extra: 0
  checks:
    DicomsegContentCheck:
      files: 1
conclusion: true

```

</details>

LennyN95 committed:
Update default.yml sample annotations

- add sample annotations for the input and output data of the default workflow
- include the IDC reference metadata (as provided in the PR)

LennyN95 committed:
Update composite.yml sample annotation

- add sample description for input and output of the composite workflow
- include the idc metadata as provided in the mhub PR (for ct and pet scans)

LennyN95 committed:
fix missing quote in default.yml workflow
